### ISSUE TITLE
Stack ignores the executable bit on ghc and then fails

### ISSUE THREAD

**Issue by johntyree** [commercialhaskell/stack#2538]
### General summary/comments (optional)

If there's a `ghc` in your path somewhere that does not have executable permission, stack will fail with `/whatever/path/ghc: streamingProcess: runInteractiveProcess: exec: permission denied (Permission denied)`

### Steps to reproduce

1. Create a file named `ghc` at the front of your `$PATH`.
2. Run command that uses ghc, e.g. `stack build`.

### Expected

I expect stack to ignore the `ghc` that is not executable, the way shells do.

### Actual

Stack tries to invoke the unexecutable `ghc` and dies.

If you suspect that a stack command misbehaved, please include the output of that command in `--verbose` mode.

```
$ stack build --verbose
Version 1.1.2 x86_64 hpack-0.14.1
2016-08-28 21:18:31.583445: [debug] Checking for project config at: /home/john/projects/brainfuck/haskell/stack.yaml @(stack-1.1.2-K10UHJ5rbiiHm4AmYlkZE:Stack.Config src/Stack/Config.hs:811:9)
2016-08-28 21:18:31.583930: [debug] Loading project config file stack.yaml @(stack-1.1.2-K10UHJ5rbiiHm4AmYlkZE:Stack.Config src/Stack/Config.hs:829:13)
2016-08-28 21:18:31.586870: [debug] Checking whether stack was built with libgmp4 @(stack-1.1.2-K10UHJ5rbiiHm4AmYlkZE:Stack.Config src/Stack/Config.hs:326:5)
2016-08-28 21:18:31.587077: [debug] Run process: ldd /usr/bin/stack @(stack-1.1.2-K10UHJ5rbiiHm4AmYlkZE:System.Process.Read src/System/Process/Read.hs:283:3)
2016-08-28 21:18:31.598180: [debug] Stack was not built with libgmp4 @(stack-1.1.2-K10UHJ5rbiiHm4AmYlkZE:Stack.Config src/Stack/Config.hs:330:14)
2016-08-28 21:18:31.598422: [debug] Trying to decode /home/john/.stack/build-plan-cache/x86_64-linux/lts-6.13.cache @(stack-1.1.2-K10UHJ5rbiiHm4AmYlkZE:Data.Binary.VersionTagged src/Data/Binary/VersionTagged.hs:55:5)
2016-08-28 21:18:31.627936: [debug] Success decoding /home/john/.stack/build-plan-cache/x86_64-linux/lts-6.13.cache @(stack-1.1.2-K10UHJ5rbiiHm4AmYlkZE:Data.Binary.VersionTagged src/Data/Binary/VersionTagged.hs:64:13)
2016-08-28 21:18:31.628848: [debug] Getting system compiler version @(stack-1.1.2-K10UHJ5rbiiHm4AmYlkZE:Stack.Setup src/Stack/Setup.hs:341:17)
2016-08-28 21:18:31.629055: [debug] Run process: ghc --info @(stack-1.1.2-K10UHJ5rbiiHm4AmYlkZE:System.Process.Read src/System/Process/Read.hs:283:3)
/home/john/projects/scripts/ghc: streamingProcess: runInteractiveProcess: exec: permission denied (Permission denied)
```

### Stack version

```
$ stack --version
Version 1.1.2 x86_64 hpack-0.14.1
```

### Method of installation

* An unofficial package repository (arch linux official packages)

**Comment by johntyree** [comment]
It *looks* like it should be doing this correctly already... https://github.com/commercialhaskell/stack/blob/63f203d7886801c9ec3c303db0c459f08687a3c8/src/System/Process/Read.hs#L359

Still investigating.

**Comment by Blaisorblade** [comment]
@johntyree That's from `master` but you're testing 1.1.2—blame leads to 633ac50f03586326ec8f70a2a2c34b9e167f4bf0 and to #2489; the fix should be in 1.2.0.
I hope closing as duplicate is OK—please reopen if this happens on branch `master` or `rc/v1.2.0`.

**Comment by johntyree** [comment]
Good call. LGTM
